src/core/dom/image/interface.ts

```diff
@@ -0,0 +1,310 @@
+ *
+ * Released under the MIT license
+ * https://github.com/V4Fire/Client/blob/master/LICENSE
+ */
+
+import iBlock from 'super/i-block/i-block';
+
+import {
+
+	SHADOW_PREVIEW_SYMBOL,
+	SHADOW_BROKEN_SYMBOL,
+	SHADOW_MAIN_SYMBOL,
+	ID_SYMBOL,
+	IMG_IS_LOADED_SYMBOL,
+	INIT_LOAD_SYMBOL,
+	LOADING_STARTED_SYMBOL
+
+} from 'core/dom/image';
+
+export interface ImageOptions {
+	/**
+	 * URL of an image
+	 */
+	src?: string;
+
+	/**
+	 * Base url for `src` and `srcset`
+	 *
+	 * @example
+	 * ```typescript
+	 * {
+	 *   src: 'img.png',
+	 *   baseSrc: 'https://url-to-img',
+	 *   ctx: this
+	 * }
+	 * ```
+	 *
+	 * ```html
+	 * <img src="https://url-to-img/img.png">
+	 * ```
+	 */
+	baseSrc?: string;
+
+	/**
+	 * Srcset of an image. This option helps to manage the situation with multiple resolutions of the image to load.
+	 * @see https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images
+	 */
+	srcset?: Dictionary<string> | string;
+
+	/**
+	 * Image `sizes` attribute
+	 * @see https://developer.mozilla.org/ru/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images
+	 */
+	sizes?: string;
+
+	/**
+	 * `source` tags for a `picture` tag
+	 * @see https://developer.mozilla.org/ru/docs/Web/HTML/Element/source
+	 */
+	sources?: ImageSource[];
+
+	/**
+	 * Alternative value of an image to improve accessibility
+	 */
+	alt?: string;
+
+	/** @see [[ImageBackgroundOptions]] */
+	bgOptions?: ImageBackgroundOptions;
+
+	/**
+	 * If true, then for each change of the image (initial, preview, broken, main)
+	 * the class will be installed with the current state
+	 *
+	 * @example
+	 * ```typescript
+	 * {
+	 *   src: 'img.png',
+	 *   stateClasses: true,
+	 *   ctx: this
+	 * }
+	 * ```
+	 *
+	 * ```html
+	 * <div class="b-block b-block__v-image_state_preview"></div>
+	 * ```
+	 *
+	 * @default `false`
+	 */
+	stageClasses?: boolean;
+
+	/**
+	 * Options for a preview image.
+	 *
+	 * The preview image will be showing while the main image is loading.
+	 */
+	preview?: string | ImageHelperOptions;
+
+	/**
+	 * Options for a broken image.
+	 *
+	 * The broken image will be showing if the loading error appears.
+	 */
+	broken?: string | ImageHelperOptions;
+
+	/**
+	 * Execution context.
+	 *
+	 * The context is used to provide a component environment, like, async, event emitters, etc.
+	 * When API is used as a directive, the context will automatically taken from a VNode instance.
+	 *
+	 * Make sure you are not using `load` or `error` without the context provided this can lead to unexpected results.
+	 *
+	 * When used as a directive, the context will be set automatically
+	 * (but you can override it explicitly by specifying the context).
+	 *
+	 * @example
+	 * ```typescript
+	 * class Test {
+	 *   setImageToDiv() {
+	 *     ImageLoader.init(this.$refs.div, {src: 'https://img.jpg', ctx: this})
+	 *   }
+	 * }
+	 * ```
+	 */
+	ctx?: iBlock;
+
+	/**
+	 * If `true` then default stages will be used
```

Comment:
Я не уверен, что в предложениях If true, нужно его оборачивать (как минимум мы в других местах так не делаем). 

```suggestion
	 * If true, then image placeholder will use the default options.
```

PS: а так ли нужен этот параметр? Почему нельзя руководствоваться простой эвристикой: если параметр задан явно, то он всегда главнее.